python_modules/dagster/dagster/core/host_representation/external_data.py

```diff
@@ -570,6 +593,22 @@ def external_asset_graph_from_defs(
     for asset_key, node_tuple_list in node_defs_by_asset_key.items():
         node_def = node_tuple_list[0][0]
         job_names = [node_tuple[1].name for node_tuple in node_tuple_list]
+
+        # temporary workaround to retrieve asset definition from job
+        output = node_def.output_dict.get("result", None)
+        op_asset_def = (
+            output.metadata.get("asset_def", None) if output and output.metadata else None
+        )
+
+        partitions_def = None
+        if op_asset_def and op_asset_def.partitions_def:
+            if isinstance(op_asset_def.partitions_def, TimeWindowPartitionsDefinition):
+                partitions_def = external_time_window_partitions_definition_from_def(
+                    op_asset_def.partitions_def
+                )
+            elif isinstance(op_asset_def.partitions_def, StaticPartitionsDefinition):
+                partitions_def = op_asset_def.partitions_def
```

Comment:
It would be helpful to raise an error if we receive a value for `partitions_def` that we don't support. Silently omitting the partitions definition could make the problem difficult to track down later on.

Reply:
Thanks--this makes sense. Adding now